network/p2p/p2pbuilder/libp2pNodeBuilder.go

```diff
@@ -318,40 +296,44 @@ func (builder *LibP2PNodeBuilder) Build() (p2p.LibP2PNode, error) {
 	node.SetUnicastManager(unicastManager)
 
 	cm := component.NewComponentManagerBuilder().
-		AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
-			// routing system is created here, because it needs to be created during the node startup.
-			routingSystem, err := builder.buildRouting(ctx, h)
-			if err != nil {
-				ctx.Throw(fmt.Errorf("could not create routing system: %w", err))
-			}
-			node.SetRouting(routingSystem)
-			builder.gossipSubBuilder.SetRoutingSystem(routingSystem)
-
-			// gossipsub is created here, because it needs to be created during the node startup.
-			gossipSub, err := builder.gossipSubBuilder.Build(ctx)
-			if err != nil {
-				ctx.Throw(fmt.Errorf("could not create gossipsub: %w", err))
-			}
-			node.SetPubSub(gossipSub)
-			gossipSub.Start(ctx)
-			ready()
-
-			<-gossipSub.Done()
-		}).
-		AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
-			// encapsulates shutdown logic for the libp2p node.
-			ready()
-			<-ctx.Done()
-			// we wait till the context is done, and then we stop the libp2p node.
-
-			err = node.Stop()
-			if err != nil {
-				// ignore context cancellation errors
-				if !errors.Is(err, context.Canceled) && !errors.Is(err, context.DeadlineExceeded) {
-					ctx.Throw(fmt.Errorf("could not stop libp2p node: %w", err))
+		AddWorker(
+			func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+				if builder.routingFactory != nil {
+					routingSystem, err := builder.routingFactory(ctx, h)
+					if err != nil {
+						ctx.Throw(fmt.Errorf("could not create routing system: %w", err))
+					}
+					node.SetRouting(routingSystem)
+					builder.gossipSubBuilder.SetRoutingSystem(routingSystem)
+				} else {
+					builder.logger.Warn().Msg("no routing system is set for the node; ignore the log for non-access and non-execution nodes")
```

Comment:
is this needed? I don't think we should log a warning on every startup for normal behavior. a missing routing system is a software bug and will have noticeable side effects during development.

Reply:
https://github.com/onflow/flow-go/pull/4800/commits/21e8489022d1377aa6fe058029dec4170227d218